### ISSUE TITLE
The `--quiet` flag doesn't silence all warnings

### ISSUE THREAD

**Issue by niklasmohrin** [dbrgn/tealdeer#268]
In fish, when typing `tldr t<Tab>`, I get additional output printed to stderr:
```plain
tldr tThe cache hasn't been updated for 34 days.
You should probably run `tldr --update` soon.

tabula          textql            tlmgr-update         trap
tac             theharvester      tlp                  trash
tail            thunar            tlp-stat             trash-cli
tailscale       thunderbird       tmpmail              travis
```

This is because we use a plain `tldr --list` for the completion. The zsh script avoids this by redirecting `2>/dev/null`, the bash one either has the same problem or bash completions automatically silence this, I haven't checked. I suppose `2>/dev/null` is okay, but I would expect `--quiet` to suffice for this and prefer using that instead.

**Comment by newsch** [comment]
I can confirm this is a problem with the bash completion as well:
```
$ source completion/bash_tealdeer
$ touch -d '31 days ago' ~/.cache/tealdeer/tldr-pages
$ tldr tThe cache hasn't been updated for 31 days.
You should probably run `tldr --update` soon.
```

FWIW, `-q` doesn't suppress the "Cache not found. Please run `tldr --update`." message.

**Comment by Walker-00** [comment]
Is this been fixed?, both "fish" and "bash" works fine for me. Shouldn't this close?

**Comment by niklasmohrin** [comment]
Completion seems to work for some other reason, but the original point that `tldr --quiet` should not display warnings or errors still stands

**Comment by sidsri14** [comment]
The remaining warning leak is fixed: the old-naming custom pages warning now respects --quiet (PR #520), with a regression test covering both modes.
